src/main/java/org/wso2/carbon/connector/operations/EmailGetBody.java

```
@@ -0,0 +1,100 @@
+                        Boolean overwriteBody) throws ConnectException {
+
+        String emailId = (String) getParameter(messageContext, EmailConstants.EMAIL_ID);
+        String folder = (String) getParameter(messageContext, EmailConstants.FOLDER);
+        EmailConnectionHandler handler = EmailConnectionHandler.getConnectionHandler();
+        String connectionName = null;
+        MailBoxConnection connection = null;
+        EmailMessage emailMessage = null;
+        try {
+            if (emailId != null) {
+                connectionName = EmailUtils.getConnectionName(messageContext);
+                connection = (MailBoxConnection) handler.getConnection(connectionName);
+                emailMessage = EmailUtils.getEmail(connection, emailId, folder);
+                if (log.isDebugEnabled()) {
+                    log.debug(format("Retrieving email body for email at id %s...", emailId));
+                }
+                
+                // Create JSON response with email details
+                JsonObject resultJSON = getResultJSON(emailMessage);
+
+                handleConnectorResponse(messageContext, responseVariable, overwriteBody, resultJSON, null, null);
+            } else {
+                JsonObject resultJSON = generateErrorResult(messageContext, Error.INVALID_CONFIGURATION);
+                handleConnectorResponse(messageContext, responseVariable, overwriteBody, resultJSON, null, null);
+                handleException(format("%s Email Id is not set.", ERROR), messageContext);
+            }
+        }
+        catch (InvalidConfigurationException e) {
+            JsonObject resultJSON = generateErrorResult(messageContext, Error.INVALID_CONFIGURATION);
+            handleConnectorResponse(messageContext, responseVariable, overwriteBody, resultJSON, null, null);
+            handleException(ERROR, e, messageContext);
+        } catch (EmailConnectionException | EmailParsingException e) {
+            throw new RuntimeException(e);
```

Shall we throw a different exception here?

---

Addressed with [e76e1d5](https://github.com/wso2-extensions/esb-connector-email/pull/39/commits/e76e1d58196aefb2e043453192cfbe6cd6ae7799).